repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,174 @@
+
+![Design](K8sRepairnatorDesign.jpg)
+
+## Prerequisites
+
+* kubectl
+* a working k8s cluster (preferable high cpu capacity)
+* gcloud (for creating the cluster mentoned above and creating mongodb if you don't have them already)
+
+## Setup ActiveMQ
+
+To deploy ActiveMQ for effective queue managements of jobs to the pipeline and scanner. First apply the yaml file inside the queue-for-buildids folder
+```
+	kubectl create -f /queue-for-buildids/activemq.yaml
+```
+To access to the web interface
+```
+	kubectl get pods 
+```
+It should look like activemq-XXXXXXX-XXXXX. Then 
+```
+	kubectl exec -it activemq-XXXXXXX-XXXXX bash
+```
+To expose to localhost for webinterface access and also publishing(sending message to queue) outside the cluster. 
+```
+	kubectl port-forward activemq-XXXXXXX-XXXXX 8161:8161 61613:61613
+```
+Access webinterface with "http://localhost:8161/admin/" in your browser. The default username is "admin" and password is also "admin".
+
+To send message to queue use the publisher.py script in queue-for-buildids folder. Syntax
+```
+	python publisher.py -d queue-name message-1 message-2 message3 ... message-N
+```
+
+queue-name is according to the format /queue/name. For instance /queue/scanner will send message to the scanner queue.
+You can use this later to manually input build id to the pipeline queue or project to the scanner queue
+
+## Setup scanners and pipelines
+
+To deploy the scanner first go into folder "scanner-dockerimage" then directly build and push the image to docker hub(enviroment variables can be specified or changed later in the "repairnator-scanner.yaml" file. With docker 
+```
+	docker build -t repairnator-scanner:tagname .
+	docker tag repairnator-scanner:tagname YOUR_DOCKERHUB_NAME/repairnator-scanner
+	docker push YOUR_DOCKERHUB_NAME/repairnator-scanner:tagname
+```
+Then go to "repairnator-deployment-yamlfiles" folder. Open repairnator-scanner.yaml then replace the image 
+
+```
+...
+	containers:
+      - name: repairnator-scanner
+        image: YOUR_DOCKERHUB_NAME/repairnator-scanner:tagname
+...
+```
+
+Now create the scanner with 
+```
+	kubectl create -f repairnator-scanner.yaml
+```
+Then go to "http://localhost:8161/admin/queues.jsp" and you should see a scanner queue with one consumer.
+
+Same goes for the pipeline, first go to "pipeline-dockerimage docker build and push it and modify the yaml file then fill in env values in the "repairnator-pipeline.yaml" file , then create with kubectl and you should also see on the same page a pipeline queue with one consumer.
+
+## Example run
+In the folder "queue-for-buildids" use publisher.py to send messages to the queue. If you have a slug (like "surli/failingProject") you can send it to the scanner queue for scanning. The scanner will auto submit the interesting BuildIds to the pipeline queue for generating patches. So sending by
+```
+	python publisher.py /queue/scanner surli/failingProject
```

Comment:
where is file `publisher.py`?

Reply:
it's included in the recent commit in the "queue-for-buildids" folder.